File: apps/api/src/schemas/stats.py
from datetime import date

from pydantic import BaseModel


class MatchStatsSummary(BaseModel):
    video_id: str
    match_id: str | None
    match_date: date | None
    opponent_name: str | None
    distance_km: float
    avg_speed_kmh: float
    max_speed_kmh: float


class AggregatedStatsOut(BaseModel):
    matches_analyzed: int
    total_distance_km: float
    avg_speed_kmh: float
    max_speed_kmh: float
    heatmap: list[list[int]]
    matches: list[MatchStatsSummary]

File: packages/shared-types/src/stats.ts
export interface MatchStatsSummary {
  video_id: string;
  match_id: string | null;
  match_date: string | null;
  opponent_name: string | null;
  distance_km: number;
  avg_speed_kmh: number;
  max_speed_kmh: number;
}

export interface AggregatedStats {
  matches_analyzed: number;
  total_distance_km: number;
  avg_speed_kmh: number;
  max_speed_kmh: number;
  heatmap: number[][];
  matches: MatchStatsSummary[];
}
